File: src/Frontend/projects/core/src/lib/models/dto/pim/attribute.ts
import { AttributeList } from './attribute-list';
import { AttributePermission } from './attribute-permission';

export class Attribute {
    public id: number;
    public name: string;
    public groupId: number;
    public type: AttributeType;
    public categoriesIds: number[] = [];
    public permissions: AttributePermission[] = [];

    // List Type
    public listId?: number;
    public list?: AttributeList;
    // Str or Text Type
    public template?: string;
    public maxLength?: number;
    public minLength?: number;
    // Num Type
    public max?: number;
    public min?: number;
    // Date Type
    public maxDate?: Date;
    public minDate?: Date;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }
        return Object.assign(
            new Attribute(),
            obj,
            {
                permissions: obj.permissions.map((p) => AttributePermission.fromJSON(p)),
                minDate: obj.minDate ? new Date(obj.minDate) : null,
                maxDate: obj.maxDate ? new Date(obj.maxDate) : null,
            },
        );
    }
}

export enum AttributeType {
    String = 1,
    Number = 2,
    Boolean = 3,
    List = 4,
    Text = 5,
    Date = 6,
}

File: src/Backend/Common/Models/Pim/Attribute.cs
using System;
using System.Collections.Generic;

namespace Company.Common.Models.Pim
{
    public class Attribute
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public DateTime CreateTime { get; set; }

        public DateTime? DeleteTime { get; set; }

        public int? CreatorId { get; set; }

        public int? DeleterId { get; set; }

        public AttributeType Type { get; set; }

        /* Str or Text Type */
        public string Template { get; set; }

        public int? MaxLength { get; set; }

        public int? MinLength { get; set; }

        /* Num Type */
        public int? Max { get; set; }

        public int? Min { get; set; }

        /* List Type */
        public int? ListId { get; set; }

        public List List { get; set; }

        /* */
        public int? GroupId { get; set; }

        /* Date Type */
        public DateTime? MinDate { get; set; }

        public DateTime? MaxDate { get; set; }

        public AttributeGroup AttributeGroup { get; set; }

        public ICollection<AttributeCategory> AttributeCategories { get; set; } = new List<AttributeCategory>();

        public ICollection<AttributeValue> AttributeValues { get; set; } = new List<AttributeValue>();

        public ICollection<AttributePermission> AttributePermissions { get; set; } = new List<AttributePermission>();

        public Attribute()
        {

        }
    }

    public enum AttributeType
    {
        String = 1,
        Number = 2,
        Boolean = 3,
        List = 4,
        Text = 5,
        Date = 6
    }
}
